src/DocumentEditor.ts

```diff
@@ -51,7 +51,12 @@ export class DocumentEditor implements vscode.Disposable {
     }
 
     public async dispose(): Promise<void> {
-        Object.keys(this.fileMap).forEach(async (key) => await fse.remove(path.dirname(key)));
+        Object.keys(this.fileMap).forEach((key) => {
+            const backupFileName = key.substring(0, key.lastIndexOf('.')) + "-backup.json";
+            fse.ensureFileSync(backupFileName)
+            fse.copySync(key, backupFileName);
+            fse.writeFileSync(key, "We do not support editing entities across sessions. Please open an entity from the tree on the left to continue. The previous changes you made here are in the corresponding back-up file.");
```

Comment:
Instead of this:
"Please open an entity from the tree on the left to continue. The previous changes you made here are in the corresponding back-up file."

How about this:
```
`Reopen the entity or view your previous changes here: ${backupFileName}`
```

It's shorter, more general (since they might've executed a mongo command instead of using the tree), and has the path to the file